### DISCUSSION TITLE
BlackHole in Google Meet using Chrome

### DISCUSSION THREAD

**Discussion by nikosmac** [ExistentialAudio/BlackHole#115] [General]
Hello, I'm trying to share audio contents in Google Meet using Chrome browser. I’ve setup Midi Audio application with the configuration that you can see in the attachments and I’ve selected the new channels in Sound application of MacOS (I’m using Catalina). I’m quite sure that the configuration is ok because it works if I create a screen rec video (with audio) using QuickTime application. But if I try to use these channels in Chrome during a meeting done with Google Meet, the BlackHole channel part of the signals is ignored (of course I’ve selected these channels also in the Google Meet audio settings).
Do you have any suggestions to fix this issue?
Thank you for your support.

<img width="640" alt="Combined input" src="https://user-images.githubusercontent.com/64214572/80119240-b9a78a00-8589-11ea-879a-999ffb2fc74e.png">
<img width="640" alt="Combined output" src="https://user-images.githubusercontent.com/64214572/80119276-c88e3c80-8589-11ea-9311-02a960bcba38.png">

**Comment by looperg4k1** [comment]
@nikosmac @devinroth Hi I have the same problem here and maybe it's a bug? I'm also trying to combine mic input & system audio with BlackHole for Google meet/Skype.(I'm using Catalina with BlackHole v0.2.6, pls see the attached setup screenshots)

I can hear my voice & system audio perfectly in QuickTime audio recording but only if I selected "High quality". I can't hear the system audio channel from playback when I'm using "Maximum quality" in QuickTime, although the loudness meter was moving up and down when recording.... same situation in Google meet & Skype, only can hear the mic-in but no system audio. Pls help

Question:
1)Why I can record the mixed audio perfectly in QuickTime only in High but not Maximum quality?
2)If I can mix the audio in QuickTime with this setup why still missing a channel(system audio) in Google meet & Skype?
(FYI, QuickTime audio recording quality: High=44100 Hz AAC audio. Maximum=44100 Hz, uncompressed Linear PCM 24-bit AIFF-C audio)
<img width="400" alt="Screenshot01" src="https://user-images.githubusercontent.com/48321679/80280511-201ddb00-8737-11ea-88c2-d58f77a3edec.png">
<img width="400" alt="Screenshot02" src="https://user-images.githubusercontent.com/48321679/80280514-23b16200-8737-11ea-9024-3497416d87fc.png">
<img width="400" alt="Screenshot03" src="https://user-images.githubusercontent.com/48321679/80280516-257b2580-8737-11ea-999b-f8abf07a8190.png">

**Comment by nikosmac** [comment]
Hello, I'd like to add further information. I tested my configuration also with Google Meet in Safari and Firefox: with both the browsers, if I select the combined input signal (Mic+BH) not only the BH is ignored (like in Chrome) but also the microphone. I hope this information can help.

**Comment by devinroth** [comment]
@nikosmac Google Meet will only use the first 2 channels in an audio device while QuickTime will use all. I suspect that is your problem. You need something to mix your mic and BlackHole together. Does that make sense?

**Comment by nikosmac** [comment]
@devinroth Thank you for your answer. To be honest I'm not expert about this topic but it sounds good. I'd like to test it. Can you suggest me a tool to mix the channels?

**Comment by devinroth** [comment]
If you are not on Catalina try Audacity. Or You can search for free daws.

**Comment by nikosmac** [comment]
Hello, thank you for the answer. Do you think that audacity can mix the channels "online"? (The need is to mix mic+blackhole during a meeting, not for a record).

I've noticed also a couple of things:
1. Selecting BlackHole in Audio Midi application you can see the 16 channels with boxes in the right side to deactivate each channel but I cannot click on them;
2. Looking at some screenshots of Soundflower I've seen that there were two options: 64ch and 2ch.

...I'm thinking at the possibility that this kind of flexibility in terms of selection of the number of channels can be implemented directly in Blackhole. What do you think about it?

Thank you for your support.

**Comment by michaelrkn** [comment]
I tested on Google Meet and a random WebRTC test website (https://online-voice-recorder.com/) in Chrome and Safari and neither could capture all 4 channels, just the first two. I'm guessing this is a general problem with using BlackHole with a web application. I love how simple BH is - is there a way to overcome this without using a DAW? I prefer web-based platforms like Meet over Zoom but can't share audio with them, and BH would be the perfect solution if the channels could be mixed rather than added. Thanks!

**Comment by luisneton** [comment]
Is this still an issue with recent versions of Chrome? I'm using the latest version of Chrome and I can use the BlackHole device during meetings, with no problems.

**Reply by devinroth** [reply]
Great. I'll close the issue.
